Sources/BraintreePayPalMessaging/BTPayPalMessagingClient.swift

```diff
@@ -0,0 +1,150 @@
+import UIKit
+import SwiftUI
+import PayPalMessages
+
+#if canImport(BraintreeCore)
+import BraintreeCore
+#endif
+
+/// Use `BTPayPalMessagingClient` to display PayPal messages to promote offers such as Pay Later and PayPal Credit to customers.
+/// - Note: This module is in beta. It's public API may change or be removed in future releases.
+public class BTPayPalMessagingClient: UIView {
+
+    // MARK: - Properties
+
+    public weak var delegate: BTPayPalMessagingDelegate?
+
+    var apiClient: BTAPIClient
+
+    // MARK: - Initializers
+
+    ///  Initializes a PayPal Messaging client.
+    /// - Parameter apiClient: The Braintree API client
+    public init(apiClient: BTAPIClient) {
+        self.apiClient = apiClient
+
+        super.init(frame: .zero)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Public Method
+
+    /// Creates a `BTPayPalMessagingClient` to be displayed to promote offers such as Pay Later and PayPal Credit to customers.
+    /// - Parameter request: an optional `BTPayPalMessagingRequest`
+    public func createView(_ request: BTPayPalMessagingRequest = BTPayPalMessagingRequest()) {
+        apiClient.fetchOrReturnRemoteConfiguration { configuration, error in
+            if let error {
+                self.delegate?.onError(self, error: error)
+                return
+            }
+
+            guard let configuration else {
+                self.delegate?.onError(self, error: BTPayPalMessagingError.fetchConfigurationFailed)
+                return
+            }
+
+            guard let clientID = configuration.json?["paypal"]["clientId"].asString() else {
+                self.delegate?.onError(self, error: BTPayPalMessagingError.payPalClientIDNotFound)
+                return
+            }
+
+            let messageData = PayPalMessageData(
+                clientID: clientID,
+                environment: configuration.environment == "production" ? .live : .sandbox,
+                amount: request.amount,
+                placement: request.placement?.placementRawValue,
+                offerType: request.offerType?.offerTypeRawValue
+            )
+
+            messageData.buyerCountry = request.buyerCountry
+
+            let messageConfig = PayPalMessageConfig(
+                data: messageData,
+                style: PayPalMessageStyle(
+                    logoType: request.logoType.logoTypeRawValue,
+                    color: request.color.messageColorRawValue,
+                    textAlignment: request.textAlignment.textAlignmentRawValue
+                )
+            )
+
+            let messageView = PayPalMessageView(config: messageConfig, stateDelegate: self, eventDelegate: self)
+            messageView.translatesAutoresizingMaskIntoConstraints = false
```

Comment:
shouldn't this be `AutoResizing` instead of `Autoresizing`?

Reply:
This is an Apple property 😄 : https://developer.apple.com/documentation/uikit/uiview/1622572-translatesautoresizingmaskintoco

Reply:
🙃 welp, we can disregard this q then haha